docs/adr/0001-extend-tosca-yaml-model.md

```diff
@@ -0,0 +1,45 @@
+# Extend TOSCA YAML model to a full in-memory model
+
+As a programmer, I want to access the complete YAML model and not be forced to traverse through all files.
```

Comment:
This does not catch what we are trying to achieve.
Steps to be done are (in this exact order):
1. Merge TServiceTemplates
- every parsed yaml file produces its own TServiceTemplate
- output of this operation shall be one unified TServiceTemplate
2. Resolve Types
- for every TNodeTemplate and TRelationshipTemplate, inject the members of their corresponding Types into the Template object (this is meant within this review by the term 'resolve')
- again, output of this operation shall be a TServiceTemplate
3. Write AccessibilityWrapper
- Supply plugins with any needed logic to operate on the model, e.g.
  - easy traversal of model, e.g. supply NodeTemplateStacks
  - ... (im sure this will grow from time to time)

Reply:
Keep in mind that TServiceTEmplate of TOSCA YAML equals to TDefinitions in TOSCA XML.

For the programmer, there should be no difference, how the enriched objects are returned.

Reply:
I always have a "Repository" in mind, which is the "AccessibilityWrapper" in your terms. The Repository can be implemented using your steps 1 and 2. It can also implemented differently. The crucial point is how to treat the QNames.